staging/src/k8s.io/controller-manager/controller/interfaces.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import "net/http"
+
+// Interface defines the base of a controller managed by a controller manager
+type Interface interface {
+	// Name returns the canonical name of the controller.
```

Comment:
It is user facing? Does it match the controllers map? Both?

Reply:
Both decisions are up to the implementation of controller manager. (1) the controller may prepend the name when logging, etc. (2) the controller manager should match both names, the point here is that the name returned here is the ground truth.

Reply:
Because the decisions belong to the controller manager, I suggest leaving the comment brief here and only define it as "canonical" name.